row-your-boat/src/main/java/com/codeaddi/row_your_boat/controller/services/view/ViewService.java

```diff
@@ -88,6 +88,27 @@ public List<String> getAllAvailableRowersForDate(String formattedDate) {
     return RowerService.getNamesByIDs(availableRowerIds, rowerClient.getAllRowers());
   }
 
+  public List<Date> getAllPastAvailableSessionsForRower(Long rowerId){
+    List<PastSessionAvailability> allPastSessionAvailabilities = availabilityClient.getAllUpcomingPastSessionAvailability();
+
+    List<PastSessionAvailability> filteredList = allPastSessionAvailabilities.stream()
+            .filter(availability -> availability.getRowerId().equals(rowerId))
+            .toList();
+
+    List<PastSession> pastSessions = availabilityClient.getAllUpcomingPastSessions();
+
+    List<Date> pastAvailability = new ArrayList<>(); // todo, this is being updated in MRC-80 with sufficient info
+    for(PastSessionAvailability pastSessionAvailability : filteredList){
+
+      for(PastSession pastSession : pastSessions){
```

Comment:
Map<Long, Date> sessionIdToDateMap = pastSessions.stream()
            .collect(Collectors.toMap(PastSession::getUpcomingSessionId, PastSession::getDate));

    // Match availabilities with session dates using the Map
    List<Date> pastAvailability = filteredList.stream()
            .map(PastSessionAvailability::getUpcomingSessionId)
            .map(sessionIdToDateMap::get)
            .filter(Objects::nonNull) // Filter out nulls in case of unmatched IDs
            .toList();

    return pastAvailability;